### DISCUSSION TITLE
Remove the blue color from the background

### DISCUSSION THREAD

**Discussion by TNNSLAB** [ImageMagick/ImageMagick#4731] [Help]
I have to desaturate the backround of this type of images. There's a lot of blue/magenta in it. Don't touch the model or apparel, just the background. 
Is ImageMagick able to do this ? Right now i'm doing it in Photoshop but it's very slow. 
Thanks
This is the original file : 
![test2](https://user-images.githubusercontent.com/46636486/150095922-1562855f-65ff-4543-96a0-a3ee2beb131f.jpg)

This is the file with the blue background desaturated:
![test2_desaturated](https://user-images.githubusercontent.com/46636486/150096702-60419905-eebd-4026-887b-320578d60647.jpg)

**Comment by fmw42** [comment]
This is one way. Unix syntax for IM 6.  I pick a point in the image in the background region and manually find its rgb color. 


```
infile="150095922-1562855f-65ff-4543-96a0-a3ee2beb131f.jpg"
color="srgb(233,233,240)"
fuzzval=10
convert "$infile" \( -clone 0 -modulate 100,0,100 \) \
\( -clone 0 -fuzz $fuzzval% -fill white -opaque "$color" -fill black +opaque white \) \
-compose over -composite \
result.png

```


![result](https://user-images.githubusercontent.com/13458218/150467231-663fa8f2-852b-4bf4-8da6-5272c257708b.png)



Or you could arbitrarily pick any point, say the top left corner, and get the color there.

```
infile="150095922-1562855f-65ff-4543-96a0-a3ee2beb131f.jpg"
point="0,0"
color=`convert "$infile" -format "%[pixel:u.p{$point}]" info:`
fuzzval=10
convert "$infile" \( -clone 0 -modulate 100,0,100 \) \
\( -clone 0 -fuzz $fuzzval% -fill white -opaque "$color" -fill black +opaque white \) \
-compose over -composite \
result2.png
```

![result2](https://user-images.githubusercontent.com/13458218/150467600-e1c44786-961b-4d0d-906f-40891b607fc4.png)
